src/index.js

```diff
@@ -267,21 +267,46 @@ function presence(str) {
 function cleanObject(obj) {
   for (const key in obj) {
     if (Object.prototype.hasOwnProperty.call(obj, key)) {
-      if (obj[key] === null) {
+      const value = obj[key];
+
+      if (value === null || isLiteralObject(value) && isEmpty(value)) {
         delete obj[key];
       }
     }
   }
   return obj;
 }
 
+function isLiteralObject(obj) {
+  return (!!obj) && (obj.constructor === Object);
+}
+
+function getDatasetInKebabCase(element) {
+  const datasetInKebabCase = {};
+
+  for (const attrName in element.attributes) {
+    if (Object.prototype.hasOwnProperty.call(element.attributes, attrName)) {
+      const attr = element.attributes[attrName];
+      
+      if (attr.name.startsWith('data-')) {
+        const keyWithoutDataPrefix = attr.name.replace('data-', '');
+
+        datasetInKebabCase[keyWithoutDataPrefix] = attr.value;
+      }
+    }
+  }
+
+  return datasetInKebabCase;
+}
+
 function eventProperties() {
   return cleanObject({
     tag: this.tagName.toLowerCase(),
     id: presence(this.id),
     "class": presence(this.className),
     page: page(),
-    section: getClosest(this, "data-section")
+    section: getClosest(this, "data-section"),
+    data: getDatasetInKebabCase(this)
```

Comment:
```suggestion
    data: this.dataset
```

https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

seams to be available accross browsers:
https://caniuse.com/?search=dataset

Reply:
Hey there!

`this.dataset` returns `camelCase`, but I'm returning them in `kebab-case`, so that the event properties reflect exactly how the element had them written, minus the `data` prefix. e.g.
 
```html
<div data-foo-bar="baz"></div>
```

Would be:

```js
properties: {
  data: {
    "foo-bar": "baz"
  }
}
```